.travis.yml

```diff
@@ -40,55 +40,129 @@ cache:
 services: mongodb
 
 before_install:
-    - stty cols 120
-    - PHP=$TRAVIS_PHP_VERSION
-    # Matrix lines for intermediate PHP versions are skipped for pull requests
-    - if [[ ! $deps && ! $PHP = ${MIN_PHP%.*} && ! $PHP = hhvm* && $TRAVIS_PULL_REQUEST != false ]]; then deps=skip; skip=1; fi
-    # A sigchild-enabled-PHP is used to test the Process component on the lowest PHP matrix line
-    - if [[ ! $deps && $PHP = ${MIN_PHP%.*} && ! -d php-$MIN_PHP/sapi ]]; then wget http://museum.php.net/php5/php-$MIN_PHP.tar.bz2 -O - | tar -xj; (cd php-$MIN_PHP; ./configure --enable-sigchild --enable-pcntl; make -j2); fi
-    - if [[ ! $PHP = hhvm* ]]; then INI_FILE=~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini; else INI_FILE=/etc/hhvm/php.ini; fi
-    - if [[ ! $skip ]]; then echo date.timezone = Europe/Paris >> $INI_FILE; fi
-    - if [[ ! $skip ]]; then echo memory_limit = -1 >> $INI_FILE; fi
-    - if [[ ! $skip ]]; then echo session.gc_probability = 0 >> $INI_FILE; fi
-    - if [[ ! $skip ]]; then echo opcache.enable_cli = 1 >> $INI_FILE; fi
-    - if [[ ! $skip ]]; then echo hhvm.jit = 0 >> $INI_FILE; fi
-    - if [[ ! $skip && $PHP = 5.* ]]; then echo extension = mongo.so >> $INI_FILE; fi
-    - if [[ ! $skip && $PHP = 5.* ]]; then echo extension = memcache.so >> $INI_FILE; fi
-    - if [[ ! $skip && $PHP = 5.* ]]; then (echo yes | pecl install -f apcu-4.0.11 && echo apc.enable_cli = 1 >> $INI_FILE); fi
-    - if [[ ! $skip && $PHP = 7.* ]]; then (echo yes | pecl install -f apcu-5.1.6 && echo apc.enable_cli = 1 >> $INI_FILE); fi
-    - if [[ ! $deps && $PHP = 5.* ]]; then (cd src/Symfony/Component/Debug/Resources/ext && phpize && ./configure && make && echo extension = $(pwd)/modules/symfony_debug.so >> $INI_FILE); fi
-    - if [[ ! $skip && $PHP = 5.* ]]; then pecl install -f memcached-2.1.0; fi
-    - if [[ ! $skip && ! $PHP = hhvm* ]]; then echo extension = ldap.so >> $INI_FILE; fi
-    - if [[ ! $skip && ! $PHP = hhvm* ]]; then phpenv config-rm xdebug.ini || echo "xdebug not available"; fi
-    - if [[ ! $skip ]]; then [ -d ~/.composer ] || mkdir ~/.composer; cp .composer/* ~/.composer/; fi
-    - if [[ ! $skip ]]; then export PHPUNIT=$(readlink -f ./phpunit); fi
+    - |
+      # General configuration
+      stty cols 120
+      PHP=$TRAVIS_PHP_VERSION
+      [ -d ~/.composer ] || mkdir ~/.composer
+      cp .composer/* ~/.composer/
+      export PHPUNIT=$(readlink -f ./phpunit)
```

Comment:
this is a bad idea, as it will put the whole Travis script in `-e` mode, breaking Travis features (see that your build output does not have the Travis failure report at the end). This is also why the Travis docs says we should never use `exit` in a script command.

If you want to use `-e`, you have to make it in a script executed through `sh`, not directly in the main commands.